### ISSUE TITLE
The variable '$ITEM' cannot be retrieved because it has not been set.

### ISSUE THREAD

**Issue by Tekz08** [jantari/LSUClient#3]
```
The variable '$ITEM' cannot be retrieved because it has not been set.
At C:\Program Files\WindowsPowerShell\Modules\LSUClient\1.0.0\LSUClient.psm1:203 char:68
+ ...  switch (Test-MachineSatisfiesDependency -DependencyKey $ITEM -Depend ...
+                                                             ~~~~~
    + CategoryInfo          : InvalidOperation: (ITEM:String) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : VariableIsUndefined
 
```

Target devices throwing the error are:

ThinkPad P51 20HH000TUS
ThinkPad P50 20EN001SUS
ThinkPad W541 20EF000JUS

All are on Win10 Pro Version 10.0.18362.175 Release 1903.

If I run it through PSRemoting or locally it passes the error and spits out a list of possible updates. However if I run it through PDQDeploy it instantly ends the command after the error is passed, even with the package set up to continue on error.

**Comment by MarkMelanson** [comment]
I am getting this Powershell error as well running locally.

Windows 10 Education 1809 10.0.17763.615 (Original Lenovo image, upgraded to 1809)
ThinkPad P70 20ESS21700

Windows 10 Professional 1803 10.0.17132.829 (Built from MS ISO)
ThinkPad T530 2394CTO

I did NOT see the error on:
Windows 10 Enterprise 2016 LTSB 10.0.14393
ThinkCentre M93p 10AAS2FG00

**Comment by MarkMelanson** [comment]
I did NOT see the error on:
 Windows 10 Enterprise 1809 10.0.17763.615
 ThinkCentre M93p 10AAS2FG00

**Comment by MarkMelanson** [comment]
I did get it work (mostly) on the ThinkPad P70. When the NIC driver is installed the network is disconnected but the script does not wait for reconnection so the script fails to download the remaining drivers.

Below is what worked for me:

Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force
Install-Module PowerShellGet -Force
Set-PSRepository -Name PSGallery -InstallationPolicy Trusted
Install-Module LSUClient -Force
Set-PSRepository -Name PSGallery -InstallationPolicy UnTrusted
$updates = Get-LSUpdate | Where { $_.Installer.Unattended }
$updates | Save-LSUpdate -Verbose
$updates | Install-LSUpdate -Verbose

**Comment by jantari** [comment]
The problem:

    The variable '$ITEM' cannot be retrieved because it has not been set.

is fixed now in Version 1.0.1 of the module which I just published, @MarkMelanson it's possible you already got this new version when you ran `Install-Module LSUClient -Force` just now.

If the NIC disconnecting during driver update is still an issue please open a seperate issue about it,
as I will close this one now.

And thanks all for providing Lenovo model numbers that's very helpful!
